Intermittent UI test failure - <SettingsAddonsTest.verifyAddonsCanBeUninstalled>

### Firebase Test Run:
https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5938816446985970612/executions/bs.676242b49f763bc9/testcases/1/test-cases
### Stacktrace:

androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'view has effective visibility=VISIBLE' doesn't match the selected view.
Expected: view has effective visibility=VISIBLE
Got: "SwitchMaterial{id=2131362264, res-name=enable_switch, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=true, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@4ccbeeb, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Enabled, input-type=0, ime-target=false, has-links=false, is-checked=true}"

at dalvik.system.VMStack.getThreadStackTrace(Native Method)
at java.lang.Thread.getStackTrace(Thread.java:1538)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:16)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:36)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:103)
at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:31)
at org.mozilla.fenix.ui.robots.SettingsSubMenuAddonsManagerAddonDetailedMenuRobot.assertAddonMenuItems(SettingsSubMenuAddonsManagerAddonDetailedMenuRobot.kt:39)
at org.mozilla.fenix.ui.robots.SettingsSubMenuAddonsManagerAddonDetailedMenuRobot.verifyCurrentAddonMenu(SettingsSubMenuAddonsManagerAddonDetailedMenuRobot.kt:19)
at org.mozilla.fenix.ui.SettingsAddonsTest$verifyAddonsCanBeUninstalled$5.invoke(SettingsAddonsTest.kt:92)
at org.mozilla.fenix.ui.SettingsAddonsTest$verifyAddonsCanBeUninstalled$5.invoke(SettingsAddonsTest.kt:24)
at org.mozilla.fenix.ui.robots.SettingsSubMenuAddonsManagerRobot$Transition.openDetailedMenuForAddon(SettingsSubMenuAddonsManagerRobot.kt:76)
at org.mozilla.fenix.ui.SettingsAddonsTest.verifyAddonsCanBeUninstalled(SettingsAddonsTest.kt:91)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
Caused by: junit.framework.AssertionFailedError: 'view has effective visibility=VISIBLE' doesn't match the selected view.
Expected: view has effective visibility=VISIBLE
Got: "SwitchMaterial{id=2131362264, res-name=enable_switch, visibility=VISIBLE, width=0, height=0, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=true, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@4ccbeeb, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Enabled, input-type=0, ime-target=false, has-links=false, is-checked=true}"

at androidx.test.espresso.matcher.ViewMatchers.assertThat(ViewMatchers.java:15)
at androidx.test.espresso.assertion.ViewAssertions$MatchesViewAssertion.check(ViewAssertions.java:15)
at androidx.test.espresso.ViewInteraction$SingleExecutionViewAssertion.check(ViewInteraction.java:10)
at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:11)
at androidx.test.espresso.ViewInteraction$2.call(ViewInteraction.java:2)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.Handler.handleCallback(Handler.java:873)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:6669)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

--- a/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt
+++ b/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsAddonsTest.kt
@@ -1,5 +1,6 @@
 package org.mozilla.fenix.ui
 
+import android.view.View
 import androidx.test.espresso.IdlingRegistry
 import org.mozilla.fenix.helpers.TestAssetHelper
 
@@ -11,12 +12,12 @@ import okhttp3.mockwebserver.MockWebServer
 import org.junit.Rule
 import org.junit.Before
 import org.junit.After
-import org.junit.Ignore
 import org.junit.Test
 import org.mozilla.fenix.R
 import org.mozilla.fenix.helpers.AndroidAssetDispatcher
 import org.mozilla.fenix.helpers.HomeActivityTestRule
 import org.mozilla.fenix.helpers.RecyclerViewIdlingResource
+import org.mozilla.fenix.helpers.ViewVisibilityIdlingResource
 import org.mozilla.fenix.ui.robots.homeScreen
 import org.mozilla.fenix.ui.robots.navigationToolbar
 
@@ -30,6 +31,7 @@ class SettingsAddonsTest {
 
     private lateinit var mockWebServer: MockWebServer
     private var addonsListIdlingResource: RecyclerViewIdlingResource? = null
+    private var addonContainerIdlingResource: ViewVisibilityIdlingResource? = null
 
     @get:Rule
     val activityTestRule = HomeActivityTestRule()
@@ -48,6 +50,10 @@ class SettingsAddonsTest {
 
         if (addonsListIdlingResource != null) {
             IdlingRegistry.getInstance().unregister(addonsListIdlingResource!!)
+        }
+
+        if (addonContainerIdlingResource != null) {
+            IdlingRegistry.getInstance().unregister(addonContainerIdlingResource!!)
         }
     }
 
@@ -89,7 +95,6 @@ class SettingsAddonsTest {
         }
     }
 
-    @Ignore("Failing intermittently on Firebase: https://github.com/mozilla-mobile/fenix/issues/13829")
     // Opens the addons settings menu, installs an addon, then uninstalls
     @Test
     fun verifyAddonsCanBeUninstalled() {
@@ -107,8 +112,13 @@ class SettingsAddonsTest {
             clickInstallAddon(addonName)
             acceptInstallAddon()
             verifyDownloadAddonPrompt(addonName, activityTestRule)
+            IdlingRegistry.getInstance().unregister(addonsListIdlingResource!!)
         }.openDetailedMenuForAddon(addonName) {
-            verifyCurrentAddonMenu()
+            addonContainerIdlingResource = ViewVisibilityIdlingResource(
+                activityTestRule.activity.findViewById(R.id.addon_container),
+                View.VISIBLE
+            )
+            IdlingRegistry.getInstance().register(addonContainerIdlingResource!!)
         }.removeAddon {
         }
     }

--- a/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAddonsManagerAddonDetailedMenuRobot.kt
+++ b/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAddonsManagerAddonDetailedMenuRobot.kt
@@ -16,8 +16,6 @@ import org.mozilla.fenix.helpers.click
 
 class SettingsSubMenuAddonsManagerAddonDetailedMenuRobot {
 
-    fun verifyCurrentAddonMenu() = assertAddonMenuItems()
-
     class Transition {
         fun goBack(interact: SettingsSubMenuAddonsManagerRobot.() -> Unit): SettingsSubMenuAddonsManagerRobot.Transition {
             fun goBackButton() = onView(allOf(withContentDescription("Navigate up")))
@@ -28,33 +26,14 @@ class SettingsSubMenuAddonsManagerAddonDetailedMenuRobot {
         }
 
         fun removeAddon(interact: SettingsSubMenuAddonsManagerRobot.() -> Unit): SettingsSubMenuAddonsManagerRobot.Transition {
+            removeAddonButton().check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
             removeAddonButton().click()
 
             SettingsSubMenuAddonsManagerRobot().interact()
             return SettingsSubMenuAddonsManagerRobot.Transition()
         }
     }
-
-    private fun assertAddonMenuItems() {
-        enableSwitchButton().check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
-        settingsButton().check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
-        detailsButton().check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
-        permissionsButton().check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
-        removeAddonButton().check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
-    }
 }
-
-private fun enableSwitchButton() =
-    onView(withId(R.id.enable_switch))
-
-private fun settingsButton() =
-    onView(withId(R.id.settings))
-
-private fun detailsButton() =
-    onView(withId(R.id.details))
-
-private fun permissionsButton() =
-    onView(withId(R.id.permissions))
 
 private fun removeAddonButton() =
     onView(withId(R.id.remove_add_on))

